code/game/machinery/doors/door.dm

```diff
@@ -459,7 +459,10 @@
 	if(!can_open(forced))
 		return
 	operating = TRUE
-	activate_mobs_in_range(src, 10)
+	var/mob/living/G
+	if(G in GLOB.player_list && G.mind.current)
+		activate_mobs_in_range(src, 15)    // Perhaps a solution to the lag. Players now activate mobs via doors.
+		// to_chat(world, "\ [G.name] successfully activated mob AI with a door.") // Debug line to make sure mobs aren't activating.
```

Comment:
```suggestion
	if(var/mob/living/carbon/human/H in oview(2))
		activate_mobs_in_range(src, 15)    // Perhaps a solution to the lag. Players now activate mobs via doors.
		// to_chat(world, "\ [H.name] successfully activated mob AI with a door.") // Debug line to make sure mobs aren't activating.
```
This is kind what you want

Reply:
Untested code that maybe works

Reply:
> This is kind what you want

Would this work with silicon that can open doors from more than 2 tiles away?